avro_schema/frontend.lua

```diff
@@ -1148,18 +1228,19 @@ Field %s is missing in source schema, and no default value was provided]],
     end
 end
 
-build_ir_error = function(offset, fmt, ...)
+build_ir_error = function(context, offset, fmt, ...)
+    local stack = context.stack
     local msg = format(fmt, ...)
-    local top, bottom = find_frames(build_ir)
     local res = {}
-    top = top + (offset or 0)
-    for i = bottom, top, -1 do
-        local _, from    = debug.getlocal(i, 1)
-        local _, to      = debug.getlocal(i, 2)
-        local _, ptrfrom = debug.getlocal(i, 5)
-        local _, ptrto   = debug.getlocal(i, 6)
-        assert(type(from) == 'table')
-        assert(type(to) == 'table')
+    local offset = offset or 0
+    for i = 1, stack.len - offset do
```

Comment:
Ugh, all that code is needed just to save two paths. It would be a way easier to read if we rework it w/ `insert(from_path, name-of-from)` before a recursive call and `remove(from_path)` after (the same for `to_path`). However doing name formatting will consume extra CPU, so maybe here fstack really gives some value.

Reply:
I suppose those approaches are equivalent

Reply:
There is already discussion for this question above.